src/kalshi.rs

```diff
@@ -21,35 +21,32 @@ fn list_questions(
     parse_response(resp)
 }
 
-pub fn get_question(client: &Client, id: &str, _config: &Settings) -> Result<KalshiMarket, KalshiError> {
-    // The Kalshi api requires the id (ticker) to be uppercase. Their frontend
-    // uses lowercase by default, but redirects given uppercase. Use
-    // uppercase to be safe. Also filter to the characters used in Kalshi
-    // tickers: alphanumeric and "-" and "."
-    let id = id.to_uppercase().chars().filter(|c| c.is_alphanumeric() || *c == '-' || *c == '.').collect::<String>();
-    let resp = client.get(format!("https://trading-api.kalshi.com/v1/events/{}/", id))
+pub fn get_question(client: &Client, input_ticker: &str, _config: &Settings) -> Result<KalshiMarket, KalshiError> {
+    // The Kalshi api requires the ticker to be uppercase, like it's given in
+    // the JSON. Their frontend uses lowercase by default, so user input is
+    // likely to need the uppercase conversion. Also filter to only the
+    // characters used in Kalshi tickers: alphanumeric and "-" and "."
+    let validated_ticker = input_ticker.to_uppercase().chars().filter(|c| c.is_alphanumeric() || *c == '-' || *c == '.').collect::<String>();
```

Comment:
I thought I commented on this in my last review, but the comment seems to have disappeared. I think it would be better to validate and return an error, rather than stripping illegal characters, e.g.
```rs
if !input_ticker.chars().all(|&c| c.is_alphanumeric() || c == '-' || c == '.') {
    return Err(KalshiError::Something);
}
```